### DISCUSSION TITLE
looking at components (paritals)

### DISCUSSION THREAD

**Discussion by mozodev** [team-durumi/theme-wwm#3] [Ideas]
https://github.com/team-durumi/theme-wwm/tree/main/layouts/partials

```
banner--admission.html
banner--section-single.html
banner--slide.html
banner--sticky.html
banner--welcome.html
breadcrumb.html
featured--courses.html
featured--exhibitions.html
featured--items.html
footer.html
form
header-home.html
header.html
```

화면을 구성요소로 분해하자고 위계 없이 쌓다보면 구성요소 파일들이 너무 많이 질 거 같...

https://tailwindui.com/

```
app | base? => 화면 전체 구조를 만드는 공통 요소들?
marketing => 뭔가 목적을 가지고 콘텐츠를 도드라지게 멋지게 보이도록 만드는 요소들?
ecommerce => 전자상거래 같은 구체적인 목적을 위한 요소들?
archive? record? ....
```

여기는  이런 식으로 큰 구분을 하는데 우리는 archive? record? 이런 디렉토리를 만들어서 쌓아볼까요?
잠시 생각이 복잡해지는 건 사실 구매 등 구체적인 행위를 위해서 필요한 UI 요소들을 제외하고 콘텐츠(혹은 상품)을 예쁘게 보여주는 목록
검색 등에 필요한 요소들은 모두 공통 요소들이라 구분을 할 때 분명한 기준을 만드는게 쉽지 않은 것 같기는 합니다.

웹사이트나 앱을 사용하는 사용자의 행위 action 는 매우 다양하겠지만 대강 생각해보면 
그냥 홈페이지, shop, 기록관리 웹사이트에서 대부분 겹치는 행위가 많을 거 같고
고유한 행위가 뭘까 고민이 되네요. 기록을 북마크? 저장하거나 메모? ㅎㅎ

요건 좀 크게 생각하는거고 ㅎㅎ 좀 더 구체적으로 내려가서 예를 들어 배너...
제 경험으로 웹사이트 배너라고 하면 종류가 다양합니다.
- 최상단 띠 배너
- 화면 중간을 덮는 배너(하나, 두개, 세개? 혹은 링크에 이미지맵 사용?)
- 화면 오른쪽에 고정으로 떠 있는 윙배너
- 메인 화면 전체를 덮는 혹은 일부 섹션에서 회전목마 이미지 배너 등
- 오늘만 보기? ㅎ

요기서 좀 더 구체적으로 회전목마 이미지 배너 => swiper를 사용했는데 
swiper container 목록이 있고 그 아래 낱장의 slide들이 여러개 들어가서 화면에 출력되죠.
이런 식으로 보면  이런 이름으로 요소를 쪼갤 수도 있겠습니다.

```
swiper--container > swiper--slide
slider--container > slider--slide 
carousel > carousel--slide
```

당장 어떻게 하자는 건 아니고 코드를 바라보며 드는 생각을 적는거에요 ㅎ
당장 작업하는데 문제는 없지만 시간을 두고 경험을 쌓아가며 천천히 생각해볼 주제인거 같아서 남깁니다.

**Comment by woonjjang** [comment]
코멘트 감사합니다. 
```
app | base? => 화면 전체 구조를 만드는 공통 요소들?
marketing => 뭔가 목적을 가지고 콘텐츠를 도드라지게 멋지게 보이도록 만드는 요소들?
ecommerce => 전자상거래 같은 구체적인 목적을 위한 요소들?
archive? record? ....
```

라이브 시켜보자고, 잘라서 구분없이 넣어놨습니다. ㅎㅎ
https://tailwindui.com/ 보면서 처음에 app/basic / marketing 나눴었는데 app-> archives-application 이라고 생각하고 app이라고 결정했던 것 같아요. 그리고 app 쪽이 아마도 썬이 자주 코멘트하는 기록 데이터랑 같이 표현되는 영역이 많이 나오는 부분일 듯

app - 만드는 프로젝트에 따라서 아카이브가 되기도 하고 ecommerce가 되기도 할 것 같아요.
| - from : 우선 데이터 모델에 관여하려는 영역이니 이쪽으로 쌓아볼께요. 
base - config.yml / menu.yml 연결되는 영역 404 등등
marketing - params.yml 연결되고, 극단적인 static 홈페이지는 base와 markting으로만 만들 수 있는 영역

partials 이름은 좀 더 시도를 ...

**Comment by woonjjang** [comment]
```
featured--courses.html
featured--exhibitions.html
featured--items.html
```
요건 데이터 모델에 featured--를 붙여본 버전

**Reply by mozodev** [reply]
그건 feature.html 파셜에 콘텐츠 유형과 표시할 개수를 넘기면 될 것 같... ㅎ
